sentry-android-ndk/src/main/jni/sentry.c

```diff
@@ -11,6 +11,90 @@ struct transport_options {
 
 struct transport_options g_transport_options;
 
+JNIEXPORT void JNICALL Java_io_sentry_android_ndk_NdkScopeObserver_nativeSetTag(JNIEnv *env, jclass cls, jstring key, jstring value) {
+    if (key == NULL) {
+        return;
+    }
+    const char *charKey = (*env)->GetStringUTFChars(env, key, 0);
+    const char *charValue = NULL;
+    if (value == NULL) {
+        charValue = (*env)->GetStringUTFChars(env, value, 0);
+    }
+    sentry_set_tag(charKey, charValue);
+}
+
+JNIEXPORT void JNICALL Java_io_sentry_android_ndk_NdkScopeObserver_nativeSetUser(
+        JNIEnv *env,
+        jclass cls,
+        jstring id,
+        jstring email,
+        jstring ipAddress,
+        jstring username) {
+    if (id == NULL && email == NULL && ipAddress == NULL && username == NULL) {
+        sentry_set_user(sentry_value_new_object());
```

Comment:
```suggestion
        sentry_remove_user();
```

Or is there really a difference to an empty user without any properties? In which cane you can just skip this early return.